src/components/views/rooms/MessageContextMenu.js

```diff
@@ -66,6 +66,16 @@ module.exports = React.createClass({
         if (this.props.onFinished) this.props.onFinished();
     },
 
+    onUnhidePreviewClick: function() {
+        if (global.localStorage) {
+            // FIXME: factor this out with LinkPreviewWidget
+            // FIXME: somehow propagate this to the EventTile such that it updates itself and realises the link has rematerialised
+            global.localStorage.removeItem("hide_preview_" + this.props.mxEvent.getId());
+        }
+        this.props.mxEvent.widgetHidden = false;
```

Comment:
I believe onFinished is sufficient as to hide the context menu the overall event tile has to be rendered